python/pyspark/sql/pandas/conversion.py

```diff
@@ -360,12 +376,25 @@ def createDataFrame(  # type: ignore[misc]
 
         assert isinstance(self, SparkSession)
 
+        timezone = self._jconf.sessionLocalTimeZone()
+
+        if type(data).__name__ == "Table":
```

Comment:
I wonder if there's a better way to do this.

Reply:
We could check if `pyarrow` exists, and maybe just `isinstance` comparison would be nicer.

Reply:
I think this is the usual method that Python developers recommend for checking the type of an object when you don't know if the module is loaded. It looks like this method is used in many other places in the PySpark code.

One improvement would be to explicitly check the module also, like this:

```py
if (type(t).__module__, type(t).__name__) == ("pyarrow.lib", "Table"):
```

I am happy to use a different approach but I am not sure which approach would be an improvement. I think the approach you suggest will not work because `isinstance` expects the name the module was loaded as (usually `pa`) but the method to check if PyArrow is loaded (`pyarrow in sys.modules`) needs the library name.

Reply:
Alternatively, I could add a separate definition for `createDataFrame` only for `data: "pa.Table"`. Let me know if you think that is preferable.

Reply:
I think it would be good to check the module also, although I'm not sure if there are any other variations to "pyarrow.lib"?

Reply:
Hmm yeah I'm not sure. Probably it's safer to add an `isinstance` assertion below as you suggest.